src/charm.py

```diff
@@ -58,20 +93,60 @@ def _on_start(self, _) -> None:
     def _mysql(self):
         """Returns an instance of the MySQL object from mysqlsh_helpers."""
         if not self._mysql_helpers:
-            # TODO: replace stubbed arguments once mechanisms to generate them exist
-            # Mechanisms = generating user/pass and storing+retrieving them from peer databag.
+            mysql_configs = self._get_or_create_mysql_configs()
+
             self._mysql_helpers = MySQL(
-                "127.0.0.1",
-                "test_cluster",
-                "password",
+                mysql_configs["unit_ip"],
+                mysql_configs["cluster_name"],
+                mysql_configs["root_password"],
                 "serverconfig",
-                "serverconfigpassword",
+                mysql_configs["server_config_password"],
                 "clusteradmin",
-                "clusteradminpassword",
+                mysql_configs["cluster_admin_password"],
             )
 
         return self._mysql_helpers
 
+    def _get_or_create_mysql_configs(self):
+        peer_relation = self.model.get_relation(PEER)
+        if peer_relation is None:
```

Comment:
nit: test with `not peer_relation`

Reply:
check no longer needed - it was removed during the refactor in 516bd9596e3be0e50d16a526485b3a330a60f2ff